libres/lib/include/ert/utils/metric.hpp

```diff
@@ -0,0 +1,25 @@
+#include <chrono>
+#include <memory>
+
+#include <ert/python.hpp>
+
+namespace ert {
+namespace utils {
+class Benchmark {
+    using clock = std::chrono::high_resolution_clock;
+    clock::time_point m_start = clock::now();
+    std::string m_message;
+    std::shared_ptr<ILogger> m_logger;
+
+public:
+    Benchmark(std::shared_ptr<ILogger> logger, const std::string &message)
+        : m_logger(logger), m_message(message) {}
+
+    ~Benchmark() {
+        m_logger->info(
+            "{} execution time: {:.4f} s", m_message,
+            std::chrono::duration<float>(clock::now() - m_start).count());
```

Comment:
When I was measuring some GPU computation this's what I had used (not sure what's the difference though):
```
auto rtime = std::chrono::duration_cast<std::chrono::microseconds>(end_time - begin_time).count()
```

Reply:
Duration is a class and duration_cast is a function that cast the incoming duration to another `(end_time - begin_time)` is implicitly a duration with a rep `unsigned int type` with a period `ratio<1,1>` aka [std::chrono::seconds](https://www.cplusplus.com/reference/chrono/seconds/)
so another way that yours could be written is 
```cpp
auto rtime = std::chrono::duration_cast<std::chrono::duration<int, std::mirco>>(end_time - begin_time).count()
//so since I wanted float and seconds, if I wanted to use duration_cast it would of been redundant example
auto rtime = std::chrono::duration_cast<std::chrono::duration<float, std::ratio<1,1>>>(end_time - begin_time).count()
```
so instead, I opted to use the class directly

Reply:
I see, the default is then the second precision which is fine I guess then :)